src/cmd/usb-util.js

```diff
@@ -73,6 +76,72 @@ class UsbPermissionsError extends Error {
 	}
 }
 
+/**
+ * Executes a function with a USB device, handling device protection and DFU mode.
+ * Given limitations from Device-OS, we currently ask the user to exit DFU mode if the device is protected.
+ *
+ * @param {Object} options - The options for executing with the USB device.
+ * @param {Object} options.args - The arguments to identify and configure the USB device.
+ * @param {Function} options.func - The function to execute with the USB device.
+ * @param {boolean} [options.dfuMode=false] - Flag indicating whether to include devices in DFU mode.
+ * @returns {Promise<*>} The result of the executed function.
+ * @throws {Error} If the device is protected and cannot be unprotected, or if the executed function throws an error.
+ *
+ * @example
+ * await executeWithUsbDevice({
+ *   args: { idOrName: 'e00fce6819ef5f971ea9563a' },
+ *   func: async (device) => {
+ *     // Perform operations with the device
+ *     return result;
+ *   },
+ *   dfuMode: true
+ * });
+ */
+async function executeWithUsbDevice({ args, func, dfuMode = false } = {}) {
```

Comment:
What is this function for? It doesn't appear to be used.

Reply:
This is the generic helper that is going to be used with `particle usb`. This is the base branch.